### PR TITLE
Alt-Svc Concepts and terminology roundup

### PR DESCRIPTION
**Pull request by LPardue**
It's easy to conflate origins and resources. So this is just me being pedantic on that. The general problem problem statement and flow of the prose is not intended to be changed.

### CODE REVIEW

File: draft-duke-httpbis-quic-version-alt-svc.md [hunk 1]
```diff
@@ -28,60 +28,70 @@ author:
...
+HTTP Alternative Services (Alt-Svc) describes how one origin's resource can be
+accessed via a different protocol/host/port combination. Alternatives are
+advertised by servers using the Alt-Svc header field or the ALTSVC frame. This
+includes a protocol name, which reuses Application Layer Protocol Negotiation
+(ALPN) codepoints. The "h3" codepoint indicates the availability of HTTP/3. A
+client that uses such an alternative first makes a QUIC connection. However,
+without a priori knowledge of which QUIC version to use, clients might incur a
+round-trip latency penalty to complete QUIC version negotiation, or forfeit
+desirable properties of a QUIC version. This document specifies a new Alt-Svc
+parameter that specifies the alternatives supported QUIC version, which
```
Comment (martinduke) [inline draft-duke-httpbis-quic-version-alt-svc.md]:
```suggestion
parameter that specifies alternative supported QUIC versions, which
```

File: draft-duke-httpbis-quic-version-alt-svc.md [hunk 2]
```diff
@@ -28,60 +28,70 @@ author:
...
+substantially reducing the chance of this penalty.
```
Comment (martinduke) [inline draft-duke-httpbis-quic-version-alt-svc.md]:
```suggestion
substantially reduces the chance of this penalty.
```

File: draft-duke-httpbis-quic-version-alt-svc.md [hunk 3]
```diff
@@ -28,60 +28,70 @@ author:
...
-UDP port, ALPN can also usefully identify the transport in an Alt-Svc context.
-The "h3" ALPN informs the client that it can use HTTP/3 {{?I-D.ietf-quic-http}}
-for access, which in turn requires the QUIC transport protocol {{?RFC8999}}.
+HTTP Alternative Services (Alt-Svc) {{!ALTSVC=I-D.ietf-httpbis-rfc7838bis}}
+describes how one origin's resource can be accessed via a different
+protocol/host/port combination. Alternatives are advertised by servers using the
+Alt-Svc header field or the ALTSVC frame. This includes a protocol name, which
+reuses codepoints from the Application-Layer Protocol Negotiation (ALPN) TLS
+extension {{?RFC7301}}. Servers can advertise multiple alternatives, in which
+case the order reflects the servers preferences (the first value being the most
```
Comment (martinduke) [inline draft-duke-httpbis-quic-version-alt-svc.md]:
```suggestion
case the order reflects the server's preferences (the first value being the most
```

File: draft-duke-httpbis-quic-version-alt-svc.md [hunk 4]
```diff
@@ -28,60 +28,70 @@ author:
...
+preferred).
+
+Clients can ignore alternative services, or pick one at their discretion. A
+client might use any details from the advertisement, in addition to out of
+band information, in determining if an alternative is suitable or preferred.
+the properties of another.
```
Comment (martinduke) [inline draft-duke-httpbis-quic-version-alt-svc.md]: ?

**Review by martinduke — COMMENTED** [review]
Thanks! Just some nits and I'll merge it.

File: draft-duke-httpbis-quic-version-alt-svc.md (hunk 4 above)
Comment (LPardue) [inline draft-duke-httpbis-quic-version-alt-svc.md]:
```suggestion
```

Comment (LPardue) [reply draft-duke-httpbis-quic-version-alt-svc.md]: whoops

**Pull request merged**